bokehjs/src/coffee/renderer/annotation/legend.coffee

```diff
@@ -83,14 +84,15 @@ class LegendView extends PlotWidget
     ctx = @plot_view.canvas_view.ctx
     ctx.save()
 
-    ctx.fillStyle = @plot_model.get('background_fill')
     @border_props.set_value(ctx)
     ctx.beginPath()
     ctx.rect(@box_coords[0], @box_coords[1],
       @legend_width, @legend_height
     )
-    ctx.fill()
     ctx.stroke()
+    # Populate background fill properties (fill_color/fill_alpha) for legend
+    @background_props.set_value(ctx)
+    ctx.fill()
```

Comment:
More context: the stroke can (potentially) overlap with the fill area, so the order can make a difference in the visual appearance. Setting a larger `line_width` would probably make it apparent. Fill-first is consistent with the way it is done everywhere else by BokehJS.